lib/internal/fs/promises.js

```diff
@@ -808,11 +808,22 @@ async function readFile(path, options) {
   return handleFdClose(readFileHandle(fd, options), fd.close);
 }
 
+async function exists(path) {
+  validateString(path, 'path');
```

Comment:
I think `access()` has already validated the path.

Reply:
Yes, but `access()` is called in the `try` block.

Reply:
https://github.com/nodejs/node/blob/73d5f8a843c89707313eb2cf2b99af59096b738a/lib/fs.js#L291-L295

`fs.existsSync()` returns `false` if path validation failed.

Reply:
According to https://github.com/nodejs/node/blob/73d5f8a843c89707313eb2cf2b99af59096b738a/lib/fs.js#L279-L284, I think we should not make the same mistake with `fsPromises.exists`.
Maybe we should even go further and also throw if the error is not ENOENT. For example a permission error now ends up in returning `false` but we actually don't know if the file exists or not.

Reply:
Currently, `fs.existsSync(path)` returns `false` when path validation failed, even path is not a `string`. e.g.

```js
> const fs = require('fs')
undefined
> fs.existsSync({})
false
``` 

I would think it is a bug(?), since [docs](https://nodejs.org/dist/latest-v16.x/docs/api/fs.html#fs_fs_existssync_path) says it only allows path of type `<string> | <Buffer> | <URL>`.

Reply:
> According to
> 
> https://github.com/nodejs/node/blob/73d5f8a843c89707313eb2cf2b99af59096b738a/lib/fs.js#L279-L284
> 
> , I think we should not make the same mistake with `fsPromises.exists`.
> Maybe we should even go further and also throw if the error is not ENOENT. For example a permission error now ends up in returning `false` but we actually don't know if the file exists or not.

If we throws, the behaivor of 3 APIs are very different:

+ `fs.exists()`: wrong callback signature;
+ `fs.existsSync()`: eats errors;
+ `fsPromise.exists()`: throws errors.

And actually they just do the same thing in different async ways. I think we should make the behaivor more chaotic.

Reply:
If you want to make `fs.existsSync` throw, we should probably make the error swallowing go through a complete deprecation first to make sure this won't break the ecosystem.